### PR TITLE
MICP - Week 5 - Homework - Palindrome Linked List

### PR DESCRIPTION
**Pull request by CaroRomo1**

### CODE REVIEW

File: Week 5/PalindromeLinkedList.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        prev = curr;
+        curr = nxt;
+    }
+    
+    return prev;
+}
+
+bool isPalindrome(ListNode *head) {
+    // Check for NULL
+    if(head != NULL || head->next != NULL) {
```
Comment (detuniko) [inline Week 5/PalindromeLinkedList.cpp]: I think this condition in if statement should be the opposite. 

So ideally, the code for input validation comes down to:
if (head == null )
    throw exception.

File: Week 5/PalindromeLinkedList.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        throw invalid_argument("NULL VALUE");
+    }
+    
+    // Check is only 1 element
```
Comment (detuniko) [inline Week 5/PalindromeLinkedList.cpp]: I really like the comments in your code. Very helpful for others who review your code!

File: Week 5/PalindromeLinkedList.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    // Compare
+    while(head != NULL && second != NULL) {
+        if(head->val != second->val) {
+            return false;
+        }
+        head = head->next;
+        second = second->next;
+    }
+    
+    return true;
```
Comment (detuniko) [inline Week 5/PalindromeLinkedList.cpp:65]: It's a good solution! I recommend you to write unit tests so you can also test your solution for the various inputs you decide in the examples part.

File: Week 5/PalindromeLinkedList.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    if(head->next == NULL){
+        return true;
+    }
+    
+    // Use two pointers to get to the center
+    ListNode *prev = head;
+    ListNode *slow = head;
+    ListNode *fast = head;
+    
+    while(fast != NULL && fast->next != NULL) {
```
Comment (alyssalingad) [inline Week 5/PalindromeLinkedList.cpp:42]: Your code is so elegant Caro! Very well written. I would not have thought of this solution - having one pointer go twice as fast to figure out where the center node is. Nice job!

File: Week 5/PalindromeLinkedList.cpp (hunk 3 above)
Comment (shiqi2015) [inline Week 5/PalindromeLinkedList.cpp:65]: I see what you mean by only reversing the half! It is smart! I am going to revise my code:p Thank you so much!

File: Week 5/PalindromeLinkedList.cpp (hunk 1 above)
Comment (CaroRomo1) [reply Week 5/PalindromeLinkedList.cpp]: Yes @detuniko ! Thank you! I upload the new correct version 😄

File: Week 5/PalindromeLinkedList.cpp (hunk 3 above)
Comment (CaroRomo1) [reply Week 5/PalindromeLinkedList.cpp:65]: Thank you @detuniko ! 😀 I will sure implement some unit tests next time :)

File: Week 5/PalindromeLinkedList.cpp (hunk 4 above)
Comment (CaroRomo1) [reply Week 5/PalindromeLinkedList.cpp:42]: Thank you very much for reviewing my code :D

File: Week 5/PalindromeLinkedList.cpp (hunk 3 above)
Comment (CaroRomo1) [reply Week 5/PalindromeLinkedList.cpp:65]: @shiqi7 Sure! Thank your for reviewing my code :D

**Pull request merged**